AppImage packaging does not work

The packaging phase of the dockerfile ends abruptly with this error message: 

> Traceback (most recent call last):
  File "/usr/local/bin/appimage-builder", line 8, in <module>
    sys.exit(__main__())
  File "/usr/local/lib/python3.9/dist-packages/appimagebuilder/__main__.py", line 44, in __main__
    invoker.execute(commands)
  File "/usr/local/lib/python3.9/dist-packages/appimagebuilder/invoker.py", line 29, in execute
    command()
  File "/usr/local/lib/python3.9/dist-packages/appimagebuilder/commands/setup_runtime.py", line 27, in __call__
    runtime.generate()
  File "/usr/local/lib/python3.9/dist-packages/appimagebuilder/modules/setup/generator.py", line 68, in generate
    self.apprun_arch.update(self._find_embed_archs(executables))
  File "/usr/local/lib/python3.9/dist-packages/appimagebuilder/modules/setup/generator.py", line 111, in _find_embed_archs
    raise RuntimeError("Unable to determine the bundle architecture")
RuntimeError: Unable to determine the bundle architecture
make: *** [Makefile:61: package-linux-appimage] Error 1

--- a/Dockerfile
+++ b/Dockerfile
@@ -4,18 +4,25 @@ RUN rustup component add rustfmt clippy \
   && sh -c "(set -x; cargo --version; rustc --version; cargo fmt --version; cargo clippy --version) > /etc/rust_toolchain_version 2>&1" \
   && apt-get update && apt-get install -y \
   cmake \
+  libxcb-render0-dev \
+  libxcb-shape0-dev \
+  libxcb-xfixes0-dev \
   libgtk-3-dev \
   curl \
   clang \
   dbus-x11 \
-  && useradd tester -d /home/tester -m
+  && useradd tester -d /app -m
 
 USER tester
+COPY --chown=tester:tester assets/ /app/assets
 
 COPY --chown=tester:tester Cargo.toml Cargo.lock /app/
 COPY --chown=tester:tester gb-bus/ /app/gb-bus/
 COPY --chown=tester:tester gb-clock/ /app/gb-clock/
 COPY --chown=tester:tester gb-cpu/ /app/gb-cpu/
+COPY --chown=tester:tester gb-apu/ /app/gb-apu/
+COPY --chown=tester:tester gb-test/ /app/gb-test/
+COPY --chown=tester:tester gb-breakpoint/ /app/gb-breakpoint/
 COPY --chown=tester:tester gb-dbg/ /app/gb-dbg/
 COPY --chown=tester:tester gb-dma/ /app/gb-dma/
 COPY --chown=tester:tester gb-joypad/ /app/gb-joypad/
@@ -27,6 +34,6 @@ COPY --chown=tester:tester gb-timer/ /app/gb-timer/
 COPY --chown=tester:tester src/ /app/src/
 
 WORKDIR /app
-RUN cargo build --workspace && cargo build --workspace --examples
+RUN cargo build --release
 
-ENTRYPOINT [ "target/debug/gbmu" ]
+ENTRYPOINT [ "target/release/gbmu" ]

--- a/Dockerfile
+++ b/Dockerfile
@@ -11,6 +11,7 @@ RUN rustup component add rustfmt clippy \
   curl \
   clang \
   dbus-x11 \
+  libasound2-dev \
   && useradd tester -d /app -m
 
 USER tester

--- a/Makefile
+++ b/Makefile
@@ -29,6 +29,10 @@ ROMS_ZIP := $(ASSETS_DIR)/roms.zip
 ROMS_DIR := $(ASSETS_DIR)/roms
 BIOS_DIR := $(ASSETS_DIR)/bios
 
+ifneq ($(OS),Windows_NT)
+UNAME_S := $(shell uname -s)
+endif
+
 requirement: roms bios
 
 bios: $(BIOS)
@@ -60,25 +64,17 @@ package-linux-appimage: docker
 	mkdir -p build
 	docker run --rm -t -v $$(pwd)/build:/build --entrypoint=/bin/sh gbmu-appimage:latest -c "set -x && appimage-builder --skip-tests && zip -r GBMU.AppDir.zip GBMU.AppDir && cp -vR GBMU-latest-x86_64.AppImage GBMU.AppDir.zip /build/"
 
-ifneq ($(OS),Windows_NT)
-	UNAME_S := $(shell uname -s)
-	ifeq ($(UNAME_S),Darwin)
 package-mac:
 	cargo build --release
 	./packaging/mac/package.sh target/release/gbmu GBMU
-endif
 
 ifeq ($(OS),Windows_NT)
 package:
 	@echo "Build on windows not supported (yet ?)"
-else
-    UNAME_S := $(shell uname -s)
-    ifeq ($(UNAME_S),Linux)
+else ifeq ($(UNAME_S),Linux)
 package: package-linux
-    endif
-    ifeq ($(UNAME_S),Darwin)
+else ifeq($(UNAME_S),Darwin)
 package: package-mac
-    endif
 endif
 
 clean:

--- a/Makefile
+++ b/Makefile
@@ -73,7 +73,7 @@ package:
 	@echo "Build on windows not supported (yet ?)"
 else ifeq ($(UNAME_S),Linux)
 package: package-linux
-else ifeq($(UNAME_S),Darwin)
+else ifeq ($(UNAME_S),Darwin)
 package: package-mac
 endif
 

--- a/packaging/linux/appimage/Dockerfile
+++ b/packaging/linux/appimage/Dockerfile
@@ -30,7 +30,7 @@ RUN chmod +x /opt/appimagetool \
   && /opt/appimagetool --appimage-extract \
   && mv -v squashfs-root appimage-tool.AppDir \
   && ln -s /opt/appimage-tool.AppDir/AppRun /usr/bin/appimagetool \
-  && pip3 install appimage-builder
+  && pip3 install appimage-builder==0.9.2
 
 RUN useradd tester -d /home/tester -m
 USER tester

--- a/packaging/linux/appimage/Dockerfile
+++ b/packaging/linux/appimage/Dockerfile
@@ -32,7 +32,7 @@ RUN chmod +x /opt/appimagetool \
   && ln -s /opt/appimage-tool.AppDir/AppRun /usr/bin/appimagetool \
   && pip3 install appimage-builder
 
-RUN useradd tester -d /home/tester -m
+RUN rm /etc/shadow.lock && useradd tester -d /home/tester -m
 USER tester
 WORKDIR /home/tester
 

--- a/packaging/linux/appimage/Dockerfile
+++ b/packaging/linux/appimage/Dockerfile
@@ -3,7 +3,7 @@ FROM gbmu:latest as BUILDER
 RUN cargo build --release
 
 # FROM appimagecrafters/appimage-builder:latest
-FROM ubuntu:21.10
+FROM ubuntu:22.04
 
 # Issue AppImage in docker: https://appimage-builder.readthedocs.io/en/latest/intro/install.html#install-appimagetool
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
@@ -25,14 +25,14 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
   zip
 
 WORKDIR /opt
-ADD https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage /opt/appimagetool
+ADD https://github.com/AppImage/AppImageKit/releases/download/13/appimagetool-x86_64.AppImage /opt/appimagetool
 RUN chmod +x /opt/appimagetool \
   && /opt/appimagetool --appimage-extract \
   && mv -v squashfs-root appimage-tool.AppDir \
   && ln -s /opt/appimage-tool.AppDir/AppRun /usr/bin/appimagetool \
   && pip3 install appimage-builder
 
-RUN rm /etc/shadow.lock && useradd tester -d /home/tester -m
+RUN useradd tester -d /home/tester -m
 USER tester
 WORKDIR /home/tester
 

